torchtune/models/llama3/_tokenizer.py

```diff
@@ -268,9 +282,22 @@ def tokenize_messages(
             if self.max_seq_len and len(tokens) >= self.max_seq_len:
                 break
 
+        # Remove final eot if add_final_eot is False
+        # and last message was from the assistant
+        if not add_final_eot:
+            if message.role == "assistant":
+                tokens = tokens[:-1]
```

Comment:
Why are we doing it this way instead of using the tokenize_message tokenize_end flag? This would still break of message eot=False. Or if a new additional end token gets added in the future.